src/ape/managers/compilers.py

```diff
@@ -45,8 +45,8 @@ def registered_compilers(self) -> Dict[str, CompilerAPI]:
         registered_compilers = {}
 
         for plugin_name, (extensions, compiler_class) in self.plugin_manager.register_compiler:
-            config = self.config.get_config(plugin_name)
-            compiler = compiler_class(config=config)
+            compiler_class.config = self.config.get_config(plugin_name)
+            compiler = compiler_class()
```

Comment:
I wonder if `config: ConfigItem` since it is contextual dependending on project (see #439) is better off staying as an instance arg